src/IDCodeValidator.java

```diff
@@ -0,0 +1,178 @@
+import java.util.*;
+
+public class IDCodeValidator {
+
+
+    public static boolean isCorrect(String idCode) {
+        System.out.println(idCode);
+        IDNumberData personalCode = IDNumberData.getInformationFromIdCode(idCode);
+        if (idCode.length() == 11
+                && isGenderNumberCorrect(personalCode.genderNumber)
+                && isDayNumberCorrect(personalCode.dayNumber)
+                && isMonthNumberCorrect(personalCode.monthNumber)
+                && isBirthDateCorrect(getFullYear(personalCode.genderNumber, personalCode.yearNumber), personalCode.monthNumber, personalCode.dayNumber)
+                && checkControlNumber(idCode)){
+
+            System.out.println("ID is correct");
+            return true;
+        }
+        if (idCode.length() != 11) {
+            System.out.println("The length is not correct");
+
+        }
+
+        return false;
+    }
+
+
+    public static boolean isGenderNumberCorrect(int genderNumber) {
+        if (genderNumber == 1 || genderNumber == 3 || genderNumber == 5 || genderNumber == 2 || genderNumber == 4 || genderNumber == 6) {
+            return true;
+        }
+        return false;
+    }
+
+
+    public static boolean isDayNumberCorrect(int dayNumber) {
+        if (dayNumber > 0 && dayNumber <= 31) {
+            System.out.println(dayNumber);
+            return true;
+        }
+        System.out.println("There is no such day");
+        return false;
+    }
+
+    public static boolean isMonthNumberCorrect(int monthNumber) {
+        if (monthNumber > 0 && monthNumber <= 12) {
+            System.out.println(monthNumber);
+            return true;
+        }
+        System.out.println("There is no such month");
+        return false;
+    }
+
+
+    public static boolean isBirthDateCorrect(int yearNumber, int monthNumber, int dayNumber) {
+        boolean result;
+        int[] numbersOfDays = {0, 31, 28, 31, 30, 31, 30, 31, 30, 31, 30, 31};
+        if (numbersOfDays[2] == 28 && isLeapYear(yearNumber)) {
```

Comment:
You don't need `numbersOfDays[2] == 28` as it's always `true`, so it can be simplified. Tip: Look at the tips from the intellij